src/components/Header/Header.tsx

```diff
@@ -0,0 +1,53 @@
+import classNames from 'classnames';
+import { LegacyRef } from 'react';
+
+type Props = {
+  handleAllCompleted: () => void;
+  allDone: boolean;
+  handleSubmit: (event: { preventDefault: () => void }) => void;
```

Comment:
React has prepared types for all the handlers so you can type it like that
`event: React.FormEvent<HTMLFormElement>`